src/app/static/src/app/components/plots/choropleth/MapAdjustScale.vue

```diff
@@ -0,0 +1,166 @@
+import {useUpdateScale} from "../useUpdateScale";
+
+export default defineComponent({
+    emits: ['update:selectedScale'],
+    props: {
+        name: {
+            type: String,
+            required: true
+        },
+        indicatorMetadata: {
+            type: Object as PropType<ChoroplethIndicatorMetadata>,
+            required: true
+        },
+        selectedScale: {
+            type: Object as PropType<ScaleSettings>,
+            required: true
+        },
+    },
+    setup(props, { emit }) {
+        const store = useStore<RootState>();
+        const scaleStep = computed(() => {
+            return store.getters["modelCalibrate/scaleStep"];
+        });
+        const {updateOutputColourScale} = useUpdateScale();
+
+        const invalidMsg = computed(() => {
+            let result = null;
+            if (props.selectedScale.type === ScaleType.Custom) {
+                if (props.selectedScale.customMin >= props.selectedScale.customMax) {
+                    result = i18next.t("maxMustBeGreaterThanMin");
+                }
+            }
+            return result;
+        });
+
+        const disableCustom = computed(() => {
+            return props.selectedScale.type !== ScaleType.Custom;
+        });
+
+        const scaleText = computed(() => {
+            if (props.indicatorMetadata) {
+                const {format, scale} = props.indicatorMetadata
+                if (!format.includes('%') && scale !== 1) {
+                    return `x ${scale}`
+                }
+            }
+            return ''
+        });
+
+        const scaleTypeGroup = computed(() => {
+            return `${props.name}-scaleType`;
+        });
+
+        const selectedScaleType = computed({
+            get() {
+                return props.selectedScale.type
+            },
+            set(newType: ScaleType) {
+                const newScaleSetting = {...props.selectedScale};
+                newScaleSetting.type = newType;
+                updateOutputColourScale(newScaleSetting)
+                emit('update:selectedScale')
+            }
+        });
+
+        const selectedScaleMin = computed({
+            get() {
+                return props.selectedScale.customMin
+            },
+            set(newMin: number) {
+                const newScaleSetting = {...props.selectedScale};
+                newScaleSetting.customMin = newMin;
+                updateOutputColourScale(newScaleSetting)
+                emit('update:selectedScale')
+            }
+        });
+
+        const selectedScaleMax = computed({
+            get() {
+                return props.selectedScale.customMax
+            },
+            set(newMax: number) {
+                const newScaleSetting = {...props.selectedScale};
+                newScaleSetting.customMax = newMax;
+                updateOutputColourScale(newScaleSetting)
+                emit('update:selectedScale')
+            }
+        });
```

Comment:
this feels weird, you get an object, than you break it apart and then you glue it back together with set functions that look the same, think you could split up the `v-model` into the `model-value` and update event and the model-value can just reference `selectedScale.<prop>` and update function can be `() => handleChange({...selectedScale, <prop>: $event})` and the `handleChange` could just do `updateOutputColourScale` and `emit`

however i think this could potentially be cleaned up by not doing commit at all, havent tested it out but if you just ditch all the emits (then your update model value function becomes `() => updateOutputColourScale({...selectedScale, <prop>: $event})`) and add a watcher to the correct part of the store that might work

Reply:
Thanks - have tidied this up after chatted through it. No need to use the emit anymore, we can have a watch in the parent on the changes to the colour scale in state